inst/sympref.m

```diff
@@ -183,140 +183,162 @@
     return
   end
 
-
-  switch lower(cmd)
-    case 'defaults'
-      settings = [];
-      settings.ipc = 'default';
-      settings.whichpython = '';
-      sympref ('display', 'default')
-      sympref ('digits', 'default')
-      sympref ('quiet', 'default')
-
-    case 'version'
-      assert (nargin == 1)
-      varargout{1} = '2.5.0-dev';
-
-    case 'display'
-      if (nargin == 1)
-        varargout{1} = settings.display;
+  if (isstruct(cmd))
+    % Means we need to copy it
+    s = sympref();
+    names = fieldnames (s);
+
+    if (sum(isfield(cmd, names)) == numfields(s))
+      if (typeinfo(cmd.ipc) == typeinfo(s.ipc) && ...
+        typeinfo(cmd.whichpython) == typeinfo(s.whichpython))
+        settings = [];
+        settings.ipc = cmd.ipc;
+        settings.whichpython = cmd.whichpython;
+        sympref ('display', cmd.display)
+        sympref ('digits', cmd.digits)
+        sympref ('quiet', cmd.quiet)
       else
-        arg = lower (arg);
-        if (strcmp (arg, 'default'))
-          arg = 'unicode';
-          if (ispc () && (~isunix ()))
-            % Unicode not working on Windows, Issue #83.
-            arg = 'ascii';
+        error ('sympref: Type of fields of the structure is not correct');
+      end
+    else
+      error ('sympref: invalid structure, cannot copy');
+    end
+
+  else
```

Comment:
Here, we'll have a much smaller diff if we *do not* use an `else` block but rather just:
````
  return
end
````